src/nupic/ntypes/Dimensions.hpp

```diff
@@ -90,15 +90,15 @@ class Dimensions : public ::std::vector<UInt>, Serializable {
    * * Specified   - We have a good dimension. We have at least one dimension.
    *                 It's not the opposite of isUnspecified()!
    *
-   * * Invalid     - Some dimension is 0.
+   * * Invalid     - Some dimension is 0 although Dontcare state is valid.
    *
    * There is a function to check for each of these states.
    */
-  bool isUnspecified() const { return(size() == 0); }
-  bool isDontcare()    const { return(size() == 1 && at(0) == 0); }
-  bool isInvalid()     const { return(getCount() == 0); }
-  bool isSpecified()   const { return(!isInvalid()); }
   static const int DONTCARE = 0;
+  bool isUnspecified() const { return(size() == 0); }
+  bool isDontcare()    const { return(size() == 1 && at(0) == DONTCARE); }
+  bool isInvalid()     const { return(size() > 1 && getCount() == 0); }
```

Comment:
add `&& !isDontcare() )` 
as in 

>   * * Invalid     - Some dimension is 0 although Dontcare state is valid.

Reply:
> add && !isDontcare() )

The `size( ) > 1` skips over the dontcare but perhaps it might be more readable as you suggest.